platform-dom/src/main/java/com/softicar/platform/dom/elements/select/value/simple/display/DomSimpleValueSelectDefaultValueComparator.java

```diff
@@ -59,22 +61,32 @@ public int compare(V first, V second) {
 
 			if (isArgumentsCastableTo(first, second, IDisplayable.class)) {
 				var comparator = new CollatingDisplayStringFunctionBasedComparator<>(displayStringFunctionSupplier.get());
-				result = Comparator.nullsFirst(comparator).compare(first, second);
+				result = compareNullsFirst(comparator, first, second);
 			}
 
 			if (result == 0) {
 				if (isArgumentsCastableTo(first, second, IBasicItem.class)) {
 					var comparator = BasicItemComparator.get();
-					result = Comparator.nullsFirst(comparator).compare((IBasicItem) first, (IBasicItem) second);
+					result = compareNullsFirst(comparator, (IBasicItem) first, (IBasicItem) second);
 				}
 			}
 
 			return result;
 		}
 
 		else {
+			if (isArgumentsCastableTo(first, second, BigDecimal.class)) {
+				return compareNullsFirst(BigDecimal::compareTo, (BigDecimal) first, (BigDecimal) second);
+			} else if (isArgumentsCastableTo(first, second, Number.class)) {
+				Number firstNumber = (Number) first;
+				Number secondNumber = (Number) second;
+				return compareNullsFirst(
```

Comment:
this is not necessary, the test `isArgumentsCastableTo` would have failed, if one of the values was `null`

Reply:
ok, that method is strange